File: cs50x/week_06/mario-more.py
from cs50 import get_int


def main():
    # prompt the user for input
    height = get_height()
    # print pyramids. for each row
    for i in range(height):
        # conditional statement. printing inverse space_pyramid and hashsigns
        for j in range(1, height, 1):
            if i + j >= height:
                print("#", end="")
            else:
                print(" ", end="")
        # printing the unchanging part
        print("#", end="")
        print(" ", end="")
        print(" ", end="")
        print("#", end="")
        # printing normal pyramid
        for k in range(height):
            if i > k:
                print("#", end="")
        print()


def get_height():
    # repeat loop for getting input
    while True:
        n = get_int("How many steps should the pyramid have?: ")
        if n >= 1 and n <= 8:
            break
    return n


main()

File: cs50x/week_01/mario-more.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int n;
  
    do
    {
        // Prompt user for size
        n = get_int("How tall should the towers be?: ");
    }
    while (n < 1 || n > 8);

    // For each row
    for (int i = 0; i < n; i++)
    {
        // For each column
        for (int j = 1; j < n; j++)
        {
            // fork here
            if (i + j >= n)
            {
                // Print a brick
                printf("#");
            }
            else
            {
                // Print a blank space
                printf(" ");
            }
        }
        printf("#");
        printf(" ");
        printf(" ");
        printf("#");
      
        for (int k = 0; k < n; k++)
        {
            if (i > k)
            {
                // Print a brick
                printf("#");
            }
        }
        printf("\n");
    }
}
